MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/Reflectometry/DataPreprocessorAlgorithm.h

```diff
@@ -0,0 +1,135 @@
+#ifndef MANTID_CUSTOMINTERFACES_DATAPREPROCESSORALGORITHM_H
+#define MANTID_CUSTOMINTERFACES_DATAPREPROCESSORALGORITHM_H
+
+#include "MantidAPI/AlgorithmManager.h"
+
+#include <set>
+#include <string>
+
+namespace MantidQt {
+namespace CustomInterfaces {
+/** @class DataPreprocessorAlgorithm
+
+DataPreprocessorAlgorithm defines a pre-processor algorithm that will be
+responsible for pre-processsing a specific column in a Data Processor UI.
+
+Copyright &copy; 2011-14 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+National Laboratory & European Spallation Source
+
+This file is part of Mantid.
+
+Mantid is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3 of the License, or
+(at your option) any later version.
+
+Mantid is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+File change history is stored at: <https://github.com/mantidproject/mantid>.
+Code Documentation is available at: <http://doxygen.mantidproject.org>
+*/
+class DataPreprocessorAlgorithm {
+public:
+  /** Constructor
+  * @param name : The name of the pre-processing algorithm
+  * @param prefix : The prefix that will added to the output workspace name
+  * @param blacklist : The list of properties we don't want to show
+  * @param show : Whether or not to show the information associated with
+  * this pre-processor in the processed workspace's name
+  */
+  DataPreprocessorAlgorithm(
+      const std::string &name, const std::string &prefix = "",
+      const std::set<std::string> &blacklist = std::set<std::string>(),
+      bool show = true)
+      : m_name(name), m_prefix(prefix), m_blacklist(blacklist), m_show(show) {
+
+    Mantid::API::IAlgorithm_sptr alg =
```

Comment:
This is a code-smell to me. You have to have registered all algorithms in order to use this, or even test it. I would have the constructor take a `const IAlgorithm&` rather than a name.